tasks/banana.js

```diff
@@ -30,62 +40,121 @@ module.exports = function ( grunt ) {
+				translatedFiles, translatedData = {}, // Translated message data
 				message,
-				documentationIndex,
+				index,
+				offset,
 				documentationMessageBlanks = [],
 				sourceMessageMissing = [],
 				count = 0;
 
-			try {
-				sourceMessages = grunt.file.readJSON( path.resolve( dir, options.sourceFile ) );
-				sourceMessageKeys = Object.keys( sourceMessages );
-			} catch ( e ) {
-				grunt.log.error( 'Loading source messages failed: "' + e + '".' );
-				ok = false;
-				return;
-			}
+			function messages( filename, type ) {
+				var messageArray;
+
+				try {
+					messageArray = grunt.file.readJSON( path.resolve( dir, filename ) );
+				} catch ( e ) {
+					grunt.log.error( 'Loading ' + type + ' messages failed: "' + e + '".' );
+					ok = false;
+					return;
+				}
 
-			try {
-				documentationMessages = grunt.file.readJSON( path.resolve( dir, options.documentationFile ) );
-				documentationMessageKeys = Object.keys( documentationMessages );
-			} catch ( e ) {
-				grunt.log.error( 'Loading documentation messages failed: "' + e + '".' );
-				ok = false;
-				return;
+				return messageArray;
 			}
+			function keysNoMetadata( messageArray, type ) {
+				var keys, offset;
 
-			sourceMessagesMetadataIndex = sourceMessageKeys.indexOf( '@metadata' );
-			if ( options.requireMetadata && sourceMessagesMetadataIndex === -1 ) {
-				grunt.log.error( 'Source file lacks a metadata block.' );
-				ok = false;
-				return;
+				try {
+					keys = Object.keys( messageArray );
+				} catch ( e ) {
+					grunt.log.error( 'Loading ' + type + ' messages failed: "' + e + '".' );
+					ok = false;
+					return;
+				}
+
+				offset = keys.indexOf( '@metadata' );
+				if ( offset === -1 ) {
+					if ( options.requireMetadata ) {
+						grunt.log.error( 'No metadata block in the ' + type + ' messages file.' );
+						ok = false;
+						return;
+					}
+				} else {
+					keys.splice( offset, 1 );
+				}
+
+				return keys;
 			}
-			sourceMessageKeys.splice( sourceMessagesMetadataIndex, 1 );
+			sourceMessages = messages( options.sourceFile, 'source' );
+			sourceMessageKeys = keysNoMetadata( sourceMessages, 'source' );
+
+			documentationMessages = messages( options.documentationFile, 'documentation' );
+			documentationMessageKeys = keysNoMetadata( documentationMessages, 'documentation' );
+
 			// Count after @metadata is removed
 			messageCount += sourceMessageKeys.length;
 
-			documentationMessagesMetadataIndex = documentationMessageKeys.indexOf( '@metadata' );
-			if ( options.requireMetadata && documentationMessagesMetadataIndex === -1 ) {
-				grunt.log.error( 'Documentation file lacks a metadata block.' );
-				ok = false;
-				return;
-			}
-			documentationMessageKeys.splice( documentationMessagesMetadataIndex, 1 );
+			translatedFiles = fs.readdirSync( dir ).filter( function ( value ) {
+				return (
+					value !== options.sourceFile &&
+					value !== options.documentationFile &&
+					value.match( new RegExp( '.*\.json' ) )
```

Comment:
Avoid RegExp constructor unless you need to embed user input. Use a literal regex instead, like `/.*.json/`. Behaves identical but makes it easier to read and may redue (double) string escaping in the future.